lib/pacemaker/pcmk_sched_primitive.c

```diff
@@ -1287,36 +1293,23 @@ DemoteRsc(pe_resource_t *rsc, pe_node_t *next, bool optional)
                      "Skipping demote of multiply active resource %s "
                      "on expected node %s",
                      rsc->id, pe__node_name(next));
-        return true;
+        return;
     }
 
     pe_rsc_trace(rsc, "%s", rsc->id);
 
-    /* CRM_CHECK(rsc->next_role == RSC_ROLE_UNPROMOTED, return FALSE); */
     for (gIter = rsc->running_on; gIter != NULL; gIter = gIter->next) {
         pe_node_t *current = (pe_node_t *) gIter->data;
 
         pe_rsc_trace(rsc, "%s on %s", rsc->id, pe__node_name(next));
         demote_action(rsc, current, optional);
     }
-    return true;
 }
 
-static bool
+static void
 RoleError(pe_resource_t *rsc, pe_node_t *next, bool optional)
 {
-    CRM_ASSERT(rsc);
-    crm_err("%s on %s", rsc->id, pe__node_name(next));
-    CRM_CHECK(false, return false);
-    return false;
-}
-
-static bool
-NullOp(pe_resource_t *rsc, pe_node_t *next, bool optional)
-{
-    CRM_ASSERT(rsc);
-    pe_rsc_trace(rsc, "%s", rsc->id);
-    return FALSE;
+    CRM_ASSERT((rsc != NULL) && false);
```

Comment:
Sorta. `CRM_ASSERT()` and ilk log the condition on failure. With just `(false)` you have to look at the code to have any idea what's going on, but with `(rsc != NULL) && false` you get a hint from just the log. I thought we did this more often but I don't see any other instances -- I might have reworked older ones to avoid it.

Reply:
Although looking at it again the `rsc != NULL` doesn't help anything, I'll drop it